org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/InvisibleProjectPreferenceChangeListener.java

```diff
@@ -37,29 +31,13 @@ public void preferencesChange(Preferences oldPreferences, Preferences newPrefere
 		if (!Objects.equals(oldPreferences.getInvisibleProjectOutputPath(), newPreferences.getInvisibleProjectOutputPath()) ||
 				!Objects.equals(oldPreferences.getInvisibleProjectSourcePaths(), newPreferences.getInvisibleProjectSourcePaths())) {
 			for (IJavaProject javaProject : ProjectUtils.getJavaProjects()) {
-				IProject project = javaProject.getProject();
-				if (ProjectUtils.isVisibleProject(project)) {
-					continue;
-				}
-				if (project.equals(ProjectsManager.getDefaultProject())) {
-					continue;
-				}
-
 				try {
-					IFolder workspaceLinkFolder = javaProject.getProject().getFolder(ProjectUtils.WORKSPACE_LINK);
-					IPath rootPath = ProjectUtils.findBelongedWorkspaceRoot(workspaceLinkFolder.getLocation());
-					if (rootPath == null) {
-						continue;
-					}
-					List<IPath> sourcePaths = InvisibleProjectImporter.getSourcePaths(newPreferences.getInvisibleProjectSourcePaths(), workspaceLinkFolder);
-					List<IPath> excludingPaths = InvisibleProjectImporter.getExcludingPath(javaProject, rootPath, workspaceLinkFolder);
-					IPath outputPath = InvisibleProjectImporter.getOutputPath(javaProject, newPreferences.getInvisibleProjectOutputPath(), true /*isUpdate*/);
-					IClasspathEntry[] classpathEntries = InvisibleProjectImporter.resolveClassPathEntries(javaProject, sourcePaths, excludingPaths, outputPath);
-					javaProject.setRawClasspath(classpathEntries, outputPath, new NullProgressMonitor());
+					InvisibleProjectImporter.updateSourcePaths(javaProject);
```

Comment:
This will always use the trigger files to update source paths, regardless of the sourcePaths preference. It's a breaking change.

Only if sourcePaths preference is explicitly changed, then need to update source paths. Otherwise, just update project output path directly via javaProject.setOutputLocation. 

See @jdneo's PR https://github.com/eclipse/eclipse.jdt.ls/pull/1767/files